#### Import the data set

In [1]:
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

len(state_covid_data)

10080

In [2]:
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

### Count up the number of unique dates in the data.

In [15]:
unique_date = []
"""
create an empty list 
"""
for i in state_covid_data[1:]:
    """
    create a for loop that loop over the state_covid_data, starting from index == 1
    nest the if function in the for loop and append the unique values in the position of first index in every element in the list
    """
    if i[0] not in unique_date:
        unique_date.append(i[0])

print(len(unique_date))

225


#### The number of unique dates in the data is 225 

### Find the first date in which the District of Columbia recorded a case.

In [33]:
for i in state_covid_data[1:]:
    """
    create a for loop to loop over every element in the data list 
    find the nested lists where the second index == District of Columbia 
    print the corresponding date value in that nested list 
    break the loop once the first occurrence is located 
    """
    if i[1] == "District of Columbia":
        dates = i[0]
        print(dates)
        break

2020-03-07


#### The first data in which the District of Columbia recorded a case is 2020-03-07

### Write a function that takes in a state name as input (e.g. "Wisconsin") and outputs the date of its first case.

In [37]:
def first_case(state_name):
    """
    define a function that takes the state name (str) as input and outputs the date 
    """
    for i in state_covid_data[1:]:
        if i[1] == state_name:
            dates = i[0]
            print(dates)
            break
    """
    create a for loop to loop over every element in the data list 
    find the nested lists where the second index == District of Columbia 
    print the corresponding date value in that nested list 
    break the loop once the first occurrence is located 
    """
    
first_case("Wisconsin")

2020-02-05


#### The first case recorded in Wisconsin is on 2020-02-05

### Bonus

In [41]:
def locate_date_1000(state_name):
    """
    create a function to output the date when the number of reported cases within the state exceeded 1000.
    create a for loop that loop over the nested lists
    convert the str to int for cases 
    if the cases > 1000, break the loop and print out the corresponding date 
    """
    for i in state_covid_data[1:]:
        if i[1] == state_name:
            i3 = int(i[3])
            if i3 > 1000:
                print(i[0])
                break

locate_date_1000("Wisconsin")

2020-03-28
